spacy/cli/debug_data.py

```diff
@@ -193,6 +194,66 @@ def debug_data(
     else:
         msg.info("No word vectors present in the package")
 
+    if "spancat" in factory_names:
+        model_labels = _get_labels_from_model(nlp, "spancat")
+        labels = [list(l.keys()) for l in gold_train_data["spancat"].values()]
+        has_low_data_warning = False
+        has_no_neg_warning = False
+        has_ws_ents_error = False
+
+        msg.divider("Span Categorization")
+        msg.info(f"{len(model_labels)} label(s)")
+
+        # Check if some model labels are not present in actual dataset
+        msg.text("Labels in train data: ")
+        for span_key, data_labels in gold_train_data["spancat"].items():
+            msg.text(
+                f"Key: {span_key}, {_format_labels(data_labels.items(), counts=True)}"
+            )
+
+        missing_labels = model_labels - set(itertools.chain(*labels))
+        if missing_labels:
+            msg.warn(
+                "Some model labels are not present in the train data. The "
+                "model performance may be degraded for these labels after "
+                f"training: {_format_labels(missing_labels)}."
+            )
+        # Check for low number of examples per label
+        for span_key, labels in gold_train_data["spancat"].items():
+            for label, count in labels.items():
+                if count <= NEW_LABEL_THRESHOLD:
+                    msg.warn(
+                        f"Low number of examples for label '{label}' in key '{span_key}' ({count})"
+                    )
+                    has_low_data_warning = True
+
+                with msg.loading("Analyzing label distribution..."):
+                    neg_docs = _get_examples_without_label(
+                        train_dataset, label, "spancat"
+                    )
+                if neg_docs == 0:
+                    msg.warn(f"No examples for texts WITHOUT new label '{label}'")
+                    has_no_neg_warning = True
+
+        if not has_low_data_warning:
+            msg.good("Good amount of examples for all labels")
+        if not has_no_neg_warning:
+            msg.good("Examples without occurrences available for all labels")
+
+        if has_low_data_warning:
+            msg.text(
+                f"To train a new entity type, your data should include at "
+                f"least {NEW_LABEL_THRESHOLD} instances of the new label",
+                show=verbose,
+            )
+        if has_no_neg_warning:
+            msg.text(
+                "Training data should always include examples of entities "
+                "in context, as well as examples without a given entity "
+                "type.",
+                show=verbose,
+            )
```

Comment:
Can you pair these with `if`/`else` for each type of warning?

Reply:
Updated in https://github.com/explosion/spaCy/pull/10070/commits/2a45f7191a03ddbf357652e94f390fb992a95e6e